### PR TITLE
Implementing GetLogs in API

### PR DESCRIPTION
**Pull request by mattlockyer**
## Issue

A continuation of https://github.com/harmony-one/harmony/pull/2038

### Unit Test Coverage

Before:

Sorry I am unsure how to run the unit tests, please DM me on Discord mattlock#2132

After:

Sorry I am unsure how to run the unit tests, please DM me on Discord mattlock#2132

## Operational Checklist

1. **Does this PR introduce backward-incompatible changes to the on-disk data structure and/or the over-the-wire protocol?**. (If no, skip to question 8.)

    **YES|NO**

2. **Describe the migration plan.**. For each flag epoch, describe what changes take place at the flag epoch, the anticipated interactions between upgraded/non-upgraded nodes, and any special operational considerations for the migration.

3. **Describe how the plan was tested.**

4. **How much minimum baking period after the last flag epoch should we allow on Pangaea before promotion onto mainnet?**

5. **What are the planned flag epoch numbers and their ETAs on Pangaea?**

6. **What are the planned flag epoch numbers and their ETAs on mainnet?**

    Note that this must be enough to cover baking period on Pangaea.

7. **What should node operators know about this planned change?**

8. **Does this PR introduce backward-incompatible changes *NOT* related to on-disk data structure and/or over-the-wire protocol?** (If no, continue to question 11.)

    **YES|NO**

9. **Does the existing `node.sh` continue to work with this change?**

10. **What should node operators know about this change?**

11. **Does this PR introduce significant changes to the operational requirements of the node software, such as >20% increase in CPU, memory, and/or disk usage?**

### CODE REVIEW

**Comment by harmony-ek** [comment]
Jenkins speaking; can one of the admins verify this patch?

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/harmony-one/harmony
 
-go 1.12
+go 1.13
```
Comment (fxfactorial) [inline go.mod]: lets not update go version yet

File: hmy/api_backend.go [hunk 2]
```diff
@@ -139,7 +139,16 @@ func (b *APIBackend) ProtocolVersion() int {
 // GetLogs ...
 func (b *APIBackend) GetLogs(ctx context.Context, blockHash common.Hash) ([][]*types.Log, error) {
 	// TODO(ricl): implement
-	return nil, nil
+	// return nil, nil
+	receipts := b.hmy.blockchain.GetReceiptsByHash(blockHash)
+	if receipts == nil {
+		return nil, nil
```
Comment (fxfactorial) [inline hmy/api_backend.go]: should return error, not just nil

File: hmy/api_backend.go [hunk 3]
```diff
@@ -139,7 +139,16 @@ func (b *APIBackend) ProtocolVersion() int {
...
+	}
+	logs := make([][]*types.Log, len(receipts))
+	for i, receipt := range receipts {
+		logs[i] = receipt.Logs
+	}
+	return logs, nil
```
Comment (fxfactorial) [inline hmy/api_backend.go]: why copy?

File: core/state_processor.go [hunk 4]
```diff
@@ -189,6 +189,10 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 		receipt.ContractAddress = crypto.CreateAddress(vmenv.Context.Origin, tx.Nonce())
 	}
 	// Set the receipt logs and create a bloom for filtering
+<<<<<<< HEAD
```
Comment (fxfactorial) [inline core/state_processor.go]: remove merge conflict markers

File: hmy/api_backend.go (hunk 3 above)
Comment (flicker-harmony) [reply hmy/api_backend.go]: Pointers?

File: hmy/api_backend.go (hunk 2 above)
Comment (mattlockyer) [reply hmy/api_backend.go]: updated in https://github.com/harmony-one/harmony/pull/2059/commits/ce78ec980f8ce096359e06bc8a2f31a71935a21c

File: hmy/api_backend.go (hunk 3 above)
Comment (mattlockyer) [reply hmy/api_backend.go]: Is it a copy? Wouldn't the receipt.Logs array be passed by reference as []*types.Log at logs[i]?

Implementation borrowed from ethereum: https://github.com/ethereum/go-ethereum/blob/d90d1db609c8d77baa422d49bd371207c06b4711/eth/api_backend.go#L177

Doesn't mean it's right.

File: core/state_processor.go (hunk 4 above)
Comment (mattlockyer) [reply core/state_processor.go]: fixed in ce78ec9

**Comment by LeoHChen** [comment]
@mattlockyer , please rebase your code.  BTW, the go version has been updated to 1.13 on master branch already.

**Comment by fxfactorial** [comment]
IIRC @flicker-harmony implemented this in another PR, so closing

**Pull request closed without merging**